resource-managers/nomad/src/main/scala/org/apache/spark/scheduler/cluster/nomad/SparkNomadJobController.scala

```diff
@@ -75,6 +83,50 @@ private[spark] class SparkNomadJobController(jobManipulator: NomadJobManipulator
     }
   }
 
+  def waitForAllocRemoval(allocId: String): Unit = {
+    do {
+      Thread.sleep(500)
+    } while (!jobManipulator.isAllocStopped(allocId))
+  }
+
+  def removeExecutors(executorIds: Seq[String]): Future[Boolean] = {
+    try {
+      executorIds
+        // executor ID == "${allocId}-${epoch}"
+        .map(executorId => executorId.replaceFirst("-\\d+$", ""))
+        // parallelize this collection so we can wait for allocs to stop concurrently
+        .par
+        .map(allocId => {
+          logInfo(s"stopping alloc $allocId")
+          jobManipulator.stopAlloc(allocId)
+          // stop all allocs and wait for them concurrently
+          executorService.submit(new Callable[String] {
```

Comment:
we need to wait for the allocations to actually stop before decreasing the executor task group count, or else Nomad will try to remove allocations we don't actually want removed